packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -192,39 +192,53 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
     }
 
     /**
-     * {@inheritDoc (ISharedObject:interface).summarize}
+     * {@inheritDoc (ISharedObject:interface).captureSummaryState}
      */
-    public summarize(fullTree: boolean = false, trackState: boolean = false): ISummaryTreeWithStats {
+    public captureSummaryState(fullTree: boolean = false): any {
         // Set _isSummarizing to true. This flag is used to ensure that we only use SummarySerializer (created below)
         // to serialize handles in this object's data. The routes of these serialized handles are outbound routes
         // to other Fluid objects.
         assert(!this._isSummarizing, 0x076 /* "Possible re-entrancy! Summary should not already be in progress." */);
         this._isSummarizing = true;
 
-        let summaryTree: ISummaryTreeWithStats;
+        let state: any;
         try {
-            const serializer = new SummarySerializer(this.runtime.channelsRoutingContext);
-            const snapshot: ITree = this.snapshotCore(serializer);
-            summaryTree = convertToSummaryTreeWithStats(snapshot, fullTree);
-            assert(this._isSummarizing, 0x077 /* "Possible re-entrancy! Summary should have been in progress." */);
+            state = this.captureSummaryStateCore(this._serializer, fullTree);
+        } finally {
+            if (state === undefined) {
+                this._isSummarizing = false;
+            }
+        }
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+        return state;
+    }
+
+    /**
+     * {@inheritDoc (ISharedObject:interface).summarizeState}
+     */
+    public async summarizeState(capture): Promise<ISummaryTreeWithStats> {
+        assert(this._isSummarizing, 0x077 /* "Possible re-entrancy! Summary should have been in progress." */);
```

Comment:
Is this okay that `captureSummaryState` sets `_isSummarizing=true` and `summarizeState` sets `_isSummarizing=false`? Is there some legitimate case where the calls are not always paired or someone could get `serializer` in between?